R/sample_predictions.R

```diff
@@ -275,23 +307,36 @@ sample_nowcast <- function(point_nowcast_matrix,
 #'   draws = 5
 #' )
 #' nowcast_draws
-sample_nowcasts <- function(point_nowcast_matrix,
-                            reporting_triangle,
-                            dispersion,
-                            draws = 1000,
-                            ...) {
+sample_nowcasts <- function(
+    point_nowcast_matrix,
+    reporting_triangle,
+    uncertainty_params,
+    draws = 1000,
+    error_model = function(pred, params) sample_nb(pred, params),
```

Comment:
here is a classic case where ... was used for a reason i.e because we only want optional args in the singular function in general. Reflexct. We don't really want functions that have more than 6 args as a rule of thumb unless we really really can't avoid it. It suggests we are trying to do way to much. 

On that note we might want to consider using a constructor function instead i.e

uncertainty_opts(error, ref_time, delay, samople) and then pass that around it we had one for negbin i.e specialsied it could have nb_unertainty_obs(ref, delay) and we assign fit and sample internally. 

I think this is likely a good idea but for another issue  so here just reflect on if we should really be adding args

Reply:
I changed all the plural functions to use ...